tests_new/test_geocaching.py

```diff
@@ -0,0 +1,178 @@
+import contextlib
```

Comment:
Please note this file. Actually the whole `tests_new` directory is probably the most important change.